mobile/src/screens/HomeScreen.tsx

```diff
@@ -14,8 +14,8 @@ const HomeScreen = () => {
 
   const connectToGame = () => {
     socket.emit("assign to room");
-    socket.once("successful assign", () => {
-      navigation.navigate("Chess");
+    socket.once("successful assign", (color: string) => {
+      navigation.navigate("Chess", { color });
```

Comment:
I think if we're going to do this, we need to register the types for the navigator here `https://github.com/TritonSE/YSC-Mobile-Application/blob/main/mobile/src/navigation/types.ts`. The repo that we got the chessboard component from has good examples of this.